src/_nebari/provider/cloud/google_cloud.py

```diff
@@ -52,19 +52,21 @@ def regions() -> Set[str]:
 
 
 @functools.lru_cache()
-def instances(region: str) -> Set[str]:
+def instances(region: str) -> set[str]:
     """Return a set of available compute instances in a region."""
     credentials, project_id = load_credentials()
     zones_client = compute_v1.services.region_zones.RegionZonesClient(
         credentials=credentials
     )
-    instances_client = compute_v1.InstancesClient(credentials=credentials)
-
-    return {
-        instance.machine_type.split("/")[-1]
-        for zone in zones_client.list(project=project_id, region=region)
-        for instance in instances_client.list(project=project_id, zone=zone.name)
-    }
+    instances_client = compute_v1.MachineTypesClient(credentials=credentials)
+    zone_list = zones_client.list(project=project_id, region=region)
+    zones = [zone for zone in zone_list]
+    instance_set: set[str] = set()
```

Comment:
Is there any reason to use type hints here but nowhere else in the variable definitions?

Reply:
because mypy wanted me to?

Reply:
Hm, I did not realize we were using mypy. I don't have a problem with using type hints but it's the first time I see we are using them within variable definitions